security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -186,11 +186,23 @@ func NewSecretManagerClient(caClient security.Client, options *security.Options)
 		caRootPath:  options.CARootPath,
 	}
 
+	go ret.refreshMetrics()
 	go ret.queue.Run(ret.stop)
 	go ret.handleFileWatch()
 	return ret, nil
 }
 
+func (sc *SecretManagerClient) refreshMetrics() {
+	t := time.NewTicker(time.Minute)
+	for range t.C {
+		if sc.cache.GetWorkload() != nil {
+			item := sc.cache.GetWorkload()
+			delay := sc.rotateTime(*item)
+			RecordCertExpiry(item.ResourceName, delay)
```

Comment:
I think we want NewDerivedGauge, so we do not need to manually run a goroutine. Can compute it during each prometheus fetch

Reply:
https://github.com/istio/pkg/blob/4fff90a48d5b5b21f7ab31a18db3b94272249b54/monitoring/monitoring.go#L189

why `NewDerivedGauge` not support labels? for now, I can just put resource name on metric name, and it will looks like as following

```console
# HELP istio_agent_default_cert_expiry_seconds The left time, in seconds, when cert chain will expire. A negative value indicates the cert is expired.
# TYPE istio_agent_default_cert_expiry_seconds gauge
istio_agent_default_cert_expiry_seconds 86242.038222406

```

the format looks wired